src/sql/workbench/services/connection/browser/connectionWidget.ts

```diff
@@ -454,13 +454,16 @@ export class ConnectionWidget extends lifecycle.Disposable {
 		let currentAuthType = this.getMatchingAuthType(selectedAuthType);
 		this._userNameInputBox.hideMessage();
 		this._passwordInputBox.hideMessage();
-		this._rememberPasswordCheckBox.checked = false;
 		this._azureAccountDropdown.hideMessage();
 		this._azureTenantDropdown.hideMessage();
 		this._tableContainer.classList.add('hide-username');
 		this._tableContainer.classList.add('hide-password');
 		this._tableContainer.classList.add('hide-azure-accounts');
 
+		if (currentAuthType !== AuthenticationType.SqlLogin) {
+			this._rememberPasswordCheckBox.checked = false;
```

Comment:
The checkbox is hidden anyways, why does it matter whether it's checked? Are we saving the value of the checkbox regardless of whether it's actually applicable or not? If so that seems like the better place to fix that - only save the value if the auth type actually supports a password.

Reply:
the purpose of this code is to reset it to default value (unchecked) when auth type changes, and next time sqllogin is selected it will be unchecked, I personally don't have a preference. my first reaction was simply remove the line, but I think it also makes sense to keep it.

regarding to your comment about 'better fix'. I am afraid that was not the problem, the problem is the checkbox that was supposed to be checked was unexpected unchecked by the line I removed.

Reply:
I guess my main issue is that what if a new auth type is added later that isn't sql login but uses the checkbox. Then we'd have the same issue for that one. Maybe never an issue but if we can avoid it now why not do so?

Can you explain why we need to reset it anyways? I would expect that when an auth type that uses it is selected that the checkbox will be updated to the saved value at that time so wouldn't this be unnecessary?

Reply:
This works too 😁